src/odemis/driver/test/smaract_test.py

```diff
@@ -649,6 +650,62 @@ def test_auto_update_function(self):
         pos_after_move = self.dev.position.value
         testing.assert_pos_almost_equal(expected_pos, pos_after_move, **COMP_ARGS)
 
+    def test_param_file_nofile(self):
+        # check if forcing a ValueError is the result of a file that does not exist
+        with self.assertRaises(ValueError):
+            CONFIG_3DOF['param_file'] = "nonexistingfile.mcs2.tsv"
+            smaract.MCS2(**CONFIG_3DOF)
+
+    def test_parse_tsv_config(self):
+        # create a temporary .mcs2.tsv file
+        file_dir = "/tmp/testfile.mcs2.tsv"
+        testfile = None
+
+        try:
+            # try to create a testfile if it is somehow not successful return the Exception
+            testfile = open(file_dir, 'w')
+        except Exception as ex:
+            logging.debug(f"unable to create testfile in {file_dir}. " + ex)
+
+        # create a dictionary with parameter identifiers and values
+        param_dict = [{"num": 1, "code": 0x0307005D, "value": "0b00000001", "type": "I32"},
+                      {"num": 1, "code": 0x0309005D, "value": "0b00001000", "type": "I64"},
+                      {"num": 1, "code": 0x0308007E, "value": "Newname", "type": "Str"}]
+
+        # add a comment line to test for
+        testfile.writelines("# commented line example\n")
+        # add an int32 type property line to test for
+        testfile.writelines(f"{param_dict[0]['num']}\t{hex(param_dict[0]['code'])}\t{param_dict[0]['value']}\t"
+                            f"{param_dict[0]['type']} # Referencing Options (Inverted Start Direction)\n")
+        # add an int64 type property line to test for
+        testfile.writelines(f"{param_dict[1]['num']}\t{hex(param_dict[1]['code'])}\t{param_dict[1]['value']}\t"
+                            f"{param_dict[1]['type']} # Referencing Options (Inverted Start Direction)\n")
+        # add a String type property line to test for
+        testfile.writelines(f"{param_dict[2]['num']}\t{hex(param_dict[2]['code'])}\t{param_dict[2]['value']}\t"
+                            f"{param_dict[2]['type']} # Referencing Options (Inverted Start Direction)\n")
+
+        CONFIG_3DOF['param_file'] = file_dir
+        dev = smaract.MCS2(**CONFIG_3DOF)
+
+        # open the target file again, this time for reading only
+        testfile = open(file_dir, 'r')
+        out_params = dev.parse_tsv_config(testfile)
+
+        # check if the output Tuple contains the right property input values
+        for index, item in enumerate(out_params.items()):
```

Comment:
I think you're lucky because the dict happens to be in the same order as it was created, but you shouldn't rely on that.
I'd suggest to iterate over param_dict, and check that `(num, code, type)` is in `out_params`, and then that the value is equal.